tests/console/btrfsmaintenance.pm

```diff
@@ -0,0 +1,77 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check btrfsmaintenance for functionality
+# - run btrfs-balance.sh and btrfs-scrub.sh
+# - Check if btrfsmaintenance-refresh.service is present and started properly
+# - Check if btrfs-scrub is scheduled
+# - Check if btrfs-balance is scheduled
+# - Checks for btrfs-related cron jobs after uninstalling btrfsmaintenance
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub btrfs_service_check {
+    my $service = $_[0];
+    # Check if the given btrfs service (e.g. scrub or balance) is enabled in one of the following methods:
+    # - via btrfsmaintenance-refresh.service
+    # - via a systemd timer service
+    # - or if known as a timer
+    if (script_run("systemctl status btrfsmaintenance-refresh.service | grep -E -v 'uninstall|none' | grep $service") == 0) {
+        record_info("btrfs-$service", "btrfs-$service enabled via btrfsmaintenance-refresh.service");
+        return 0;
+    } elsif (script_run("systemctl is-enabled btrfs-$service.timer | grep enabled") == 0) {
+        record_info("btrfs-$service", "btrfs-$service enabled via btrfs-$service.timer");
+        return 0;
+    } elsif (script_run("systemctl list-timers --all | grep btrfs-$service") == 0) {
+        record_info("btrfs-$service", "btrfs-$service found in list-timers");
+        return 0;
+    } else {
+        record_info("btrfs-$service", "btrfs-$service is not enabled");
+        return 1;
+    }
+}
+
+sub run {
+    # Preparation
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    if (script_run('mount | grep btrfs') != 0) {
+        record_info("btrfs-maintenance", "No btrfs volume mounted");
+    }
+    # Run balance and scrub
+    assert_script_run('/usr/share/btrfsmaintenance/btrfs-balance.sh', timeout => 300);
+    assert_script_run('/usr/share/btrfsmaintenance/btrfs-scrub.sh ',  timeout => 300);
+    assert_script_run('systemctl restart btrfsmaintenance-refresh.service');
+    # Ensure btrfsmaintenance script is enabled
+    assert_script_run("systemctl is-enabled btrfsmaintenance-refresh | grep enabled");
+    # Check if btrfs-scrub and btrfs-balance are (somehow) enabled (results only in a info write)
+    record_info("btrfs-scrub service not active")   if (btrfs_service_check("scrub") != 0);
```

Comment:
The reason for not using `assert_script_run` was that the evaluation would fail on SLES<15 and thus breaking the test. Originally the idea was to circumvent this issue by not failing but using `record_info`.

I changed this behavior by skipping this test on SLE<15 and failing on all other variants. This might be the better solution and going in the direction you are suggesting here. This might be the better approach than my previous solution.

    # Check if btrfs-scrub and btrfs-balance are (somehow) enabled (results only in a info write)
    if (!is_sle('<15')) {
        die("btrfs-scrub service not active")   if btrfs_service_unavailable("scrub");
        die("btrfs-balance service not active") if btrfs_service_unavailable("balance");
    }

New test runs are scheduled.